### DISCUSSION TITLE
Fava Import by APIs, rather than by File

### DISCUSSION THREAD

**Discussion by mescanne** [beancount/fava#2053] [Ideas]
I have two importers into Fava -- e.g. https://pypi.org/project/starling-beancount/ and one I've written myself -- which are API driven, rather than by uploaded files.

I'd like to have a button (across multiple accounts, ideally) that would trigger an import for all of the "latest" transactions.

Right now this can be hacked by having a file (that is never moved) that you "extract", and it will fetch the latest transactions for imports.

I can take a look into changing the UI -- if there are any ideas, guidance, much appreciated.

Thanks!
Mark

**Comment by tarioch** [comment]
I use the file approach quite a bit. The nice benefits of this is that it's also the perfect place for putting credentials. Not sure if you really need something else. What is missing?
